service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapperTest.java

```diff
@@ -149,46 +146,130 @@ public void setUp() throws IOException {
+    void When_MappingObservationJson_Expect_CompoundStatementXmlOutput(String inputJson, String outputXml) {
+        final Observation observationAssociatedWithSpecimen = getObservationResourceFromJson(inputJson);
+        final String expectedXml = getXmlStringFromFile(outputXml);
 
-        String jsonInput = ResourceTestFileUtils.getFileContent(inputJson);
-        Observation observationAssociatedWithSpecimen = new FhirParseService().parseResource(jsonInput, Observation.class);
-        String expectedXmlOutput = ResourceTestFileUtils.getFileContent(outputXml);
+        when(randomIdGeneratorService.createNewId())
+            .thenReturn("random-unmapped-id");
 
-        String compoundStatementXml = observationMapper.mapObservationToCompoundStatement(
-            observationAssociatedWithSpecimen
-        );
+        final String actualXml = observationMapper.mapObservationToCompoundStatement(
+            observationAssociatedWithSpecimen);
 
-        assertThat(compoundStatementXml).isEqualToIgnoringWhitespace(expectedXmlOutput);
+        assertThat(actualXml).isEqualToIgnoringWhitespace(expectedXml);
     }
 
     @Test
-    public void When_MappingDefaultObservationJson_Expect_DefaultObservationStatementXmlOutput() throws IOException {
-        String jsonInput = ResourceTestFileUtils.getFileContent(
-            OBSERVATION_TEST_FILE_DIRECTORY + "input_default_observation.json"
-        );
-        Observation observationAssociatedWithSpecimen = new FhirParseService().parseResource(jsonInput, Observation.class);
-        String expectedXmlOutput = ResourceTestFileUtils.getFileContent(
-            OBSERVATION_TEST_FILE_DIRECTORY + "expected_output_default_observation.xml"
-        );
+    void When_MappingDefaultObservationJson_Expect_DefaultObservationStatementXmlOutput() {
+        final Observation observationAssociatedWithSpecimen =
+            getObservationResourceFromJson("input_default_observation.json");
+        final String expectedXml = getXmlStringFromFile("expected_output_default_observation.xml");
+
+        final String actualXml = observationMapper.mapObservationToCompoundStatement(
+            observationAssociatedWithSpecimen);
+
+        assertThat(actualXml).isEqualTo(expectedXml);
+    }
+
+    @Test
+    void When_MappingTestGroupHeader_With_NopatMetaSecurity_Expect_ConfidentialityCodeWithinCompoundStatement() {
+        final Observation observation = getObservationResourceFromJson("observation_associated_with_specimen_1.json");
+        ConfidentialityCodeUtility.appendNopatSecurityToMetaForResource(observation);
+
+        when(confidentialityService.generateConfidentialityCode(observation))
+            .thenReturn(Optional.of(NOPAT_CONFIDENTIALITY_CODE));
+        final String actualXml = observationMapper
+            .mapObservationToCompoundStatement(observation);
+
+        assertThat(actualXml).contains(NOPAT_CONFIDENTIALITY_CODE);
+    }
```

Comment:
We need to introduce a way to assert that the `confidentialityCode` is indeed within a `CompoundStatement`, rather than just asserting that it is present within the output XML. The same goes for the tests below, `ObservationStatement` and `NarrativeStatement`.